docs/style.md

```diff
@@ -0,0 +1,213 @@
+# InSpec style guide
+
+This is a set of recommended InSpec rules you should use when writing controls.
+
+## Control files
+
+### 1. All controls should be located in the "controls" directory and end in ".rb"
+
+Reason: Most syntax highlighters will render InSpec files correctly across a wide list of tools.
+
+Avoid: `controls/ssh_config`
+Use: `controls/ssh_config.rb`
+
+Avoid: `controls/ssh/config.rb`
+Use: `controls/ssh_config.rb`
+
+### 2. Avoid using "controls" or "control" in the name of your control files
+
+Reason: Using `controls` in the filename again duplicates it and creates unnecessary clutter when reading it. Keep the names short and concise.
+
+Avoid: `controls/ssh_controls.rb`
+Use: `controls/ssh.rb`
+
+
+## Code style
+
+### 3. Use Ruby style conventions
+
+Reason: There are many opinions on how to style code across languages and frameworks. To reduce the differences and make all code more easily readable and understandable we choose to follow some common Ruby style guidelines. The following list is a subset of the current best practices. Since InSpec is focused on writing tests instead of arbitrary programming code we can focus on a subset that is important to test authors.
+
+This is a subset of style conventions taken from [the Ruby style guide](https://github.com/rubocop-hq/ruby-style-guide):
+
+- Use: UTF-8 as source file encoding
+- Use: 2 spaces per indentation level; no hard tabs
+- Use: Unix-style file endings
+- Avoid: multiple expressions per line
+```ruby
+puts 'foo'; puts 'bar'   # Avoid
+
+puts 'foo'               # Good
+puts 'bar'
+```
+- Use: Spaces around operators, after commas, colons, and semicolons
+```ruby
+sum = 1 + 2
+```
+- Avoid: Spaces after `(`, `[` or before `]`, `)`. Use spaces around `{` and before `}`.
+```ruby
+# bad
+some( arg ).other
+[ 1, 2, 3 ].each{|e| puts e}
+
+# good
+some(arg).other
+[1, 2, 3].each { |e| puts e }
+```
+- Avoid: several empty lines in a row
+
+Please refer to the full guide for the recommended style conventions.
+
+
+### 4. Avoid unnecessary parentheses in matchers
+
+Adding additional parentheses is not required and provides more readability if it is not used:
+
+Avoid: `it { should eq(value) }`
+Use: `it { should eq value }`
+
+The exception are matchers that require additional arguments or named arguments.
+
+
+## Controls
+
+### 5. Do not wrap controls in conditional statements
+
+Reason: This will create dynamic profiles whose controls depend on the execution. The problem here is that we cannot render the profile or provide its information before scanning a system. We want to be able to inform users of the contents of their profiles before they run them. It is valid to skip controls that are not necessary for a system, as long as you do it via `only_if` conditions. Ruby's internal conditionals will hide parts of the profile to static analysis and should thus be avoided.
```

Comment:
There's a similar issue if tests are wrapped inside an iterator

e.g. to test all files under /var/log

```
  command('find /var/log -type f').stdout.split("\n").each do |log_file|
    describe file(log_file) do
      it { should_not be_writable.by('group') }
      it { should_not be_executable.by('group') }
      it { should_not be_readable.by('other') }
      it { should_not be_writable.by('other') }
      it { should_not be_executable.by('other') }
    end
  end
```

This causes a "Control has no tests defined" warning from `inspec check`.

It would be awesome if there was some Inspec concept that replaced `each`. Or if `inspec check` could assume that the iterator is sensible and not attempt to evaluate it.

Reply:
I raised https://github.com/inspec/inspec/issues/3348 just this week regarding this - we currently can't fail on `inspec check` warnings in CI due to this warning being unavoidable.